gstd/src/msg/async.rs

```diff
@@ -61,6 +62,15 @@ impl<D: Decode> FusedFuture for CodecMessageFuture<D> {
     }
 }
 
+/// To interrupt a program execition waiting for a reply on a previous message,
+/// one need to specify an `.await` expression.
+/// The initial message that requires a reply is sent instantly.
+/// methods `send_and_wait_for_reply` retun `MessageFuture` where `.await` can be specified.
+/// `MessageFuture` checks if the reply exists already:
+///     if not, program interrupts
+///     if reply exists, function returns the payload `Result`
+///     if reply came without paylod, program  with `ContractError::ExitCode`
```

Comment:
```suggestion
///     if reply exists, function checks it's exit code:
///          if it equals 0, returns `Ok(payload)`
///          else returns `Err(ContractError::ExitCode(code))`
```
```suggestion
///     if reply exists, function returns the payload `Result`
///     if reply came without paylod, program  with `ContractError::ExitCode`
```

note: Anyway need to fmt it